File: src/main/java/com/rimlang/rim/syntax/BooleanExpression.java
package com.rimlang.rim.syntax;

import com.rimlang.rim.lexer.Token;

import java.util.List;
import java.util.stream.Collectors;

public class BooleanExpression extends Expression {

    private final List<Token> tokens;

    public BooleanExpression(List<Token> tokens) {
        this.tokens = tokens;

    }

    private BooleanExpression parse(List<Token> tokens) {
        // a and (b and c)
        return null;
    }

    @Override
    public String translate() {
        return "true";//tokens.stream().map(t -> t.value().toString()).collect(Collectors.joining(" "));
    }
}

File: src/main/java/com/rimlang/rim/syntax/BooleanExpression.kt
package com.rimlang.rim.syntax

import com.rimlang.rim.lexer.Token
import com.rimlang.rim.translation.Context

data class BooleanExpression(val tokens: List<Token>) : Expression() {
    init {
        println(this)
    }
    private fun parse(tokens: List<Token>): BooleanExpression {
        // example: a and b and c    ->    a and (b and c)
        TODO("Not yet implemented")
    }

    override fun translate(context: Context): String {
        return "true"
    }
}